packages/kbn-apm-synthtrace/src/cli/utils/get_service_urls.ts

```diff
@@ -74,15 +75,53 @@ async function getKibanaUrl({ target, logger }: { target: string; logger: Logger
   }
 }
 
-export async function getServiceUrls({ logger, target, kibana }: RunOptions & { logger: Logger }) {
+export async function getServiceUrls({
+  logger,
+  target,
+  kibana,
+  pr,
+}: RunOptions & { logger: Logger }) {
+  if (pr !== undefined) {
+    // Target CI deployment in PR
+    let prNumber: string | undefined;
+    if (pr.length) {
+      // If pr is specified, extract the number
+      const regexResult = /\d+$/.exec(pr);
+      prNumber = regexResult?.[0];
+    } else {
+      // If pr is not specified, use github cli to find pr number for current branch
+      const stdout = await exec(`gh pr view`);
+      const regexResult = /^number:\s*(\d+)$/m.exec(stdout);
+      prNumber = regexResult?.[1];
```

Comment:
what about something like `@octokit` that doesn't require the user to have `gh` installed on their machine?